master is building dirty

https://release.rook.io/master/ shows dirty release versions for the past few builds.  Looking into jenkin's local repo, it looks like a change is being introduced during the build due to a `gofmt` issue or an empty line with an indentation.

another tricky part here is that `go fmt` allows package wildcards (e.g. `github.com/rook/rook/pkg/...`) while `gofmt` expects fully resolved paths.  After updating `golang.mk` to use `gofmt` directly, I get:
```
> build/run make go.fmt
stat github.com/rook/rook/cmd/...: no such file or directory
stat github.com/rook/rook/pkg/...: no such file or directory
build/makelib/golang.mk:180: recipe for target 'go.fmt' failed
```

--- a/build/makelib/golang.mk
+++ b/build/makelib/golang.mk
@@ -176,7 +176,7 @@ go.vet:
 
 .PHONY: go.fmt
 go.fmt:
-	@$(GOHOST) fmt $(GO_ALL_PACKAGES)
+	@gofmt_out=$$(gofmt -d -e $(GO_SUBDIRS) 2>&1) && [ -z "$${gofmt_out}" ] || (echo "$${gofmt_out}" 1>&2; exit 1)
 
 go.validate: go.vet go.fmt
 

--- a/pkg/operator/mon/mon.go
+++ b/pkg/operator/mon/mon.go
@@ -449,7 +449,7 @@ func validNode(node v1.Node) bool {
 	if node.Spec.Unschedulable {
 		return false
 	}
-	
+
 	// a node cannot be tainted
 	for _, t := range node.Spec.Taints {
 		if t.Effect == v1.TaintEffectNoSchedule || t.Effect == v1.TaintEffectPreferNoSchedule {

